proto/proto/penumbra/core/dex/v1alpha1/dex.proto

```diff
@@ -111,16 +111,35 @@ message SwapClaimPlan {
 }
 
 message SwapView {
+  oneof swap_view {
+    VisibleSwapView visible = 1;
+    OpaqueSwapView opaque = 2;
+  }
+}
+
+message VisibleSwapView {
   dex.v1alpha1.Swap swap = 1;
   dex.v1alpha1.SwapPlaintext swap_plaintext = 3;
 }
 
+message OpaqueSwapView {
+  dex.v1alpha1.Swap swap = 1;
+}
```

Comment:
This could be folded into the `SwapView` as a nested message, like

```
message SwapView {
  message Opaque { ... }
  message Visible { ... }
  oneof swap_view {
    Visible visible = 1;
    Opaque opaque = 2;
  }
}
```

Reply:
there's an example of this here: https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.custody.v1alpha1#penumbra.custody.v1alpha1.PreAuthorization

Reply:
that's what I was looking for, thank you!

Reply:
I'm still having trouble with the type Tetris here - I'm unsure how to refactor the from & try_from code to take the new proto implementation into account. @hdevalence given the above implementation of the proto you provided as an example, can you check out the conversions in OutputView and see if there is a better way to do those?

Reply:
Sorry, updated!